R/cube-dims.R

```diff
@@ -170,3 +170,52 @@ is.selectedDimension <- function (dims) {
     names(selecteds) <- dims@names
     return(selecteds)
 }
+
+
+#' Get dimension type
+#'
+#' This function returns the specific type of each cube dimension. This is useful
+#' when cubes contain categorical array or multiple response variables because it
+#' identifies the dimensions of the cube which refer to the different parts of
+#' array variable:
+#' - `ca_items`: Categorical array items
+#' - `ca_categories`: The categories of the categorical array
+#' - `mr_items`: Multiple response options or items
+#' - `mr_selections`: The selection status for a multiple response variable
+#'
+#' @return A character vector. This is identical to `types()` except that
+#' the array variable types are more specific.
+#' @param x a CrunchCube or CubeDims object
+#'
+#' @return a character vector of dimension types
+#' @export
+#' @keywords internal
+getDimType <-  function (x) {
```

Comment:
I don't have a super strong opinions about this, but is there a reason to use this implementation instead of https://github.com/Crunch-io/crplyr/pull/2/files#diff-7797b6031d801b050ee3c519dc43bea6R87 did I miss something?

Reply:
We chatted a bit in standup, but the major reason was to unify as much of the cube dimension typing as possible under one roof.